test/disjoint_pool.cpp

```diff
@@ -0,0 +1,107 @@
+// Copyright (C) 2023 Intel Corporation
+// Under the Apache License v2.0 with LLVM Exceptions. See LICENSE.TXT.
+// SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+
+#include "memoryPool.hpp"
+#include "pool.hpp"
+#include "provider_null.h"
+#include "provider_trace.h"
+#include "provider.hpp"
+#include "pool_disjoint_impl.hpp"
+#include "pool/pool_disjoint.h"
+
+static usm::DisjointPool::Config poolConfig() {
+    usm::DisjointPool::Config config{};
+    config.SlabMinSize = 4096;
+    config.MaxPoolableSize = 4096;
+    config.Capacity = 4;
+    config.MinBucketSize = 64;
+    return config;
+}
+
+struct umf_disjoint_pool_params params = {
```

Comment:
I would suggest making a function that returns `umf_disjoint_pool_params` rather than a global variable. `freeErrorPropagation` test modifies some members of the `params` and if we add more tests in the future, the behavior might be unexpected.

Reply:
Done